frontend/src/app/account/profile/profile.component.ts

```diff
@@ -0,0 +1,241 @@
+            check: () => this.formUtils.passwordHasNoSpecial(this.editPasswordForm),
+        },
+        { msg: 'Minimum 8 characters', check: () => this.formUtils.isPasswordShort(this.editPasswordForm) },
+    ];
+
+    ngOnInit() {
+        this.user = this.authenticationService.userValue;
+        this.showEditProfile = false;
+        this.showEditPassword = false;
+        this.isProcessingEdit = false;
+        this.isProcessingPassword = false;
+    }
+
+    onUpdateProfile() {
+        this.showEditProfile = true;
+        this.showEditPassword = false;
+    }
+
+    onUpdatePassword() {
+        this.showEditPassword = true;
+        this.showEditProfile = false;
+    }
+
+    onEditSubmit() {
+        if (this.editProfileForm.valid) {
+            this.isProcessingEdit = true;
+
+            // Check if password is correct first
+            this.authenticationService
+                .login(this.user!.username, this.editProfileForm.get('password')?.value)
+                .subscribe({
+                    next: () => {
+                        this.authenticationService
+                            .updateAccount(
+                                this.editProfileForm.get('username')?.value,
+                                this.editProfileForm.get('email')?.value,
+                                this.editProfileForm.get('password')?.value,
+                            )
+                            .subscribe({
+                                next: () => {
+                                    this.router.navigateByUrl('/account', { skipLocationChange: true }).then(() => {
+                                        this.router.navigate(['/account/profile']);
+                                    });
+                                },
+                                error: error => {
+                                    this.isProcessingEdit = false;
+                                    const status = error.cause.status;
+                                    let errorMessage = 'An unknown error occurred';
+                                    if (status === 400) {
+                                        errorMessage = 'Missing Fields';
+                                    } else if (status === 401) {
+                                        errorMessage = 'Unauthorized. Invalid/Expired JWT Token';
+                                    } else if (status === 403) {
+                                        errorMessage = 'Forbidden Access';
+                                    } else if (status === 404) {
+                                        errorMessage = 'User ID Not Found';
+                                    } else if (status === 409) {
+                                        errorMessage = 'Username or Email already exists';
+                                    } else if (status === 500) {
+                                        errorMessage = 'Internal Server Error';
+                                    }
```

Comment:
I believe some of these errors are not possible except through adversarial POSTMAN requests? I think we could cut 400 and 403.

401 and 404 are rather weird, but given this is the frontend, if we handle this, perhaps it should just be user friendly. Maybe something like a recommendation to log out and log in instead of technical details?